prometheus_client/openmetrics/parser.py

```diff
@@ -141,37 +170,121 @@ def _parse_labels(it, text):
                 break
             else:
                 raise ValueError("Invalid line: " + text)
-    return labels
+    return labels, labels_len
+
+
+def _parse_labels(text):
+    labels = {}
+    # Return if we don't have valid labels
+    if "=" not in text:
+        return labels
+
+    # Copy original labels
+    sub_labels = text
+    try:
+        # Process one label at a time
+        while sub_labels:
+            # The label name is before the equal
+            value_start = sub_labels.index("=")
+            label_name = sub_labels[:value_start]
+            sub_labels = sub_labels[value_start + 1:]
+
+            # Find the first quote after the equal
+            quote_start = sub_labels.index('"') + 1
+            value_substr = sub_labels[quote_start:]
+
+            # Check for empty label name
+            if len(label_name) == 0:
+                raise ValueError
+
+            # Check for extra commas
+            if label_name[0] == ',' or value_substr[-1] == ',':
+                raise ValueError
+
+            # Find the last unescaped quote
+            i = 0
+            while i < len(value_substr):
+                i = value_substr.index('"', i)
+                if not _is_character_escaped(value_substr[:i], i):
+                    break
+                i += 1
+
+            # The label value is inbetween the first and last quote
+            quote_end = i + 1
+            label_value = sub_labels[quote_start:quote_end]
+            # Replace escaping if needed
+            if "\\" in label_value:
+                label_value = _replace_escaping(label_value)
+            labels[label_name] = label_value
+
+            # Remove the processed label from the sub-slice for next iteration
+            sub_labels = sub_labels[quote_end + 1:]
+            next_comma = sub_labels.find(",") + 1
+            sub_labels = sub_labels[next_comma:]
+
+            # Check for missing commas
+            if sub_labels and next_comma == 0:
+                raise ValueError
+            
+        return labels
+
+    except ValueError:
+        raise ValueError("Invalid labels: " + text)
 
 
 def _parse_sample(text):
-    name = []
-    value = []
+    # Detect the labels in the text
+    try:
+        # `index` method raises a ValueError with
+        # `substring not found` message if text 
+        # doesn't contain label braces
+        label_start = text.index("{")
+        # The name is before the labels
+        name = text[:label_start]
+        seperator = " # "
+        if not name.endswith("_bucket") or text.count(seperator) == 0:
+            # Line doesn't contain an exemplar
```

Comment:
What if it (incorrectly) does?

Reply:
changed it to `if text.count(seperator) == 0:`
it should garantee that 👍